### PR TITLE
Use Table dataset to speed up training

### PR DESCRIPTION
**Pull request by shendiaomo**
This version speed up the 1st epoch by 3~4 times. Continue digging further methods.

### CODE REVIEW

File: python/sqlflow_submitter/tensorflow/input_fn.py [hunk 1]
```diff
@@ -117,43 +108,34 @@ def input_fn(select,
...
+def parse_pai_dataset(feature_column_names, has_label, feature_specs, *row):
+    features = {}
+    for i, name in enumerate(feature_column_names):
+        spec = feature_specs[name]
+        f = read_feature(row[i], spec, name)
+        features[name] = tf.SparseTensor(*f) if spec["is_sparse"] else list(f)
+    return features, row[-1] if has_label else features
+
+
+def pai_dataset(table,
```
Comment (typhoonzero) [inline python/sqlflow_submitter/tensorflow/input_fn.py:135]: Please take a look at https://github.com/sql-machine-learning/sqlflow/pull/1878/files#diff-f1dc5fd65c894da64814798d3954cebaL116, the parsing function need to deal with many data input formats.

Comment (shendiaomo) [reply python/sqlflow_submitter/tensorflow/input_fn.py:135]: The `read_feature` function is handling those input formats. Any omission？

Comment (typhoonzero) [reply python/sqlflow_submitter/tensorflow/input_fn.py:135]: I see. That should be correct, I'll test this PR using the internal CI before merging.

**Comment by typhoonzero** [comment]
Error reported when running the internal CI:

```
sqlflow_submitter/tensorflow/predict.py", line 274, in pred
    pai_table)
  File "/apsara/TempRoot/Odps/alifin_jtest_dev_20200302023210873ghwo8vzt2_3b91959f_84fe_4c12_84f9_ce40496e7eed_AlgoTask_0_0/tensorflow@bd011019049059.eu126#0/tensorflow_jobs/sqlflow_submitter/tensorflow/predict.py", line 145, in estimator_predict
    predict_generator = pai_maxcompute_db_generator(
NameError: global name 'pai_maxcompute_db_generator' is not defined
```

**Review by typhoonzero — CHANGES_REQUESTED** [review]
Please fix those errors in PAI maxcompute CI cases before merging this.

**Comment by shendiaomo** [comment]
> Please fix those errors in PAI maxcompute CI cases before merging this.

Done.

**Comment by typhoonzero** [comment]
Still got errors:

```
sqlflow_submitter/tensorflow/predict.py", line 274, in pred
    pai_table)
  File "/apsara/TempRoot/Odps/alifin_jtest_dev_20200303023706529gjgp8vzt2_5270064f_aa7c_4444_95d6_6129d3148249_AlgoTask_0_0/tensorflow@bd011019050126.eu126#0/tensorflow_jobs/sqlflow_submitter/tensorflow/predict.py", line 208, in estimator_predict
    for features in predict_generator:
  File "/apsara/TempRoot/Odps/alifin_jtest_dev_20200303023706529gjgp8vzt2_5270064f_aa7c_4444_95d6_6129d3148249_AlgoTask_0_0/tensorflow@bd011019050126.eu126#0/tensorflow_jobs/sqlflow_submitter/db.py", line 255, in reader
    selected_cols = copy.copy(feature_column_names)
NameError: global name 'copy' is not defined
```

Please run the `TestEnd2EndMaxComputePAI` mannually before merge.

File: python/sqlflow_submitter/db.py [hunk 2]
```diff
@@ -155,22 +156,21 @@ def read_feature(raw_val, feature_spec, feature_name):
         values = np.ones([indices.size], dtype=np.int32)
         dense_shape = np.array(feature_spec["shape"], dtype=np.int64)
         return (indices, values, dense_shape)
-    else:
+    elif feature_spec["delimiter"] != "":
```
Comment (typhoonzero) [inline python/sqlflow_submitter/db.py:159]: The condition `feature_spec["delimiter"] != ""` is not mutually exclusive with the one of `if feature_spec["is_sparse"]` which makes this condition hard to read. I think keep the previous  code may be easy to understand.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**